drivers/bme280/bme280.c

```diff
@@ -0,0 +1,415 @@
+
+    /* Read the uncompensated pressure */
+    int32_t adc_P = (((uint32_t)measurement_regs[0 + 0]) << 12) |
+        (((uint32_t)measurement_regs[0 + 1]) << 4) |
+        ((((uint32_t)measurement_regs[0 + 2]) >> 4) & 0x0F);
+
+    int64_t var1;
+    int64_t var2;
+    int64_t p_acc;
+
+    /*
+     * Compensate the pressure value.
+     * The following is code from Bosch's BME280_driver bme280_compensate_pressure_int64()
+     * The variable names and the many defines have been modified to make the code
+     * more readable.
+     */
+    var1 = ((int64_t)t_fine) - 128000;
+    var2 = var1 * var1 * (int64_t)cal->dig_P6;
+    var2 = var2 + ((var1 * (int64_t)cal->dig_P5) << 17);
+    var2 = var2 + (((int64_t)cal->dig_P4) << 35);
+    var1 = ((var1 * var1 * (int64_t)cal->dig_P3) >> 8) + ((var1 * (int64_t)cal->dig_P2) << 12);
+    var1 = (((((int64_t)1) << 47) + var1)) * ((int64_t)cal->dig_P1) >> 33;
+    /* Avoid division by zero */
+    if (var1 == 0) {
+        return UINT32_MAX;
+    }
+
+    p_acc = 1048576 - adc_P;
+    p_acc = (((p_acc << 31) - var2) * 3125) / var1;
+    var1 = (((int64_t)cal->dig_P9) * (p_acc >> 13) * (p_acc >> 13)) >> 25;
+    var2 = (((int64_t)cal->dig_P8) * p_acc) >> 19;
+    p_acc = ((p_acc + var1 + var2) >> 8) + (((int64_t)cal->dig_P7) << 4);
+
+    return p_acc >> 8;
+}
+
+uint16_t bme280_read_humidity(bme280_t *dev)
+{
+    assert(!dev);
+
+    bme280_calibration_t *cal = &dev->calibration;      /* helper variable */
+
+    /* Read the uncompensated pressure */
+    int32_t adc_H = (((uint32_t)measurement_regs[6 + 0]) << 8) |
+        (((uint32_t)measurement_regs[6 + 1]));
+
+    /*
+     * Compensate the humidity value.
+     * The following is code from Bosch's BME280_driver bme280_compensate_humidity_int32()
+     * The variable names and the many defines have been modified to make the code
+     * more readable.
+     * The value is first computed as a value in %rH as unsigned 32bit integer
+     * in Q22.10 format(22 integer 10 fractional bits).
+     */
+    int32_t var1;
+
+    /* calculate x1*/
+    var1 = (t_fine - ((int32_t)76800));
+    /* calculate x1*/
+    var1 = (((((adc_H << 14) - (((int32_t)cal->dig_H4) << 20) - (((int32_t)cal->dig_H5) * var1)) +
+              ((int32_t)16384)) >> 15) *
+            (((((((var1 * ((int32_t)cal->dig_H6)) >> 10) *
+                 (((var1 * ((int32_t)cal->dig_H3)) >> 11) + ((int32_t)32768))) >> 10) +
+               ((int32_t)2097152)) * ((int32_t)cal->dig_H2) + 8192) >> 14));
+    var1 = (var1 - (((((var1 >> 15) * (var1 >> 15)) >> 7) * ((int32_t)cal->dig_H1)) >> 4));
+    var1 = (var1 < 0 ? 0 : var1);
+    var1 = (var1 > 419430400 ? 419430400 : var1);
+    /* First multiply to avoid losing the accuracy after the shift by ten */
+    return (100 * ((uint32_t)var1 >> 12)) >> 10;
+}
+
+/* Read compensation data, 0x88..0x9F, 0xA1, 0xE1..0xE7 */
+static int read_calibration_data(bme280_t* dev)
+{
```

Comment:
Because it is only called from bme280 where dev is already used, and thus must be valid.

Hmm. Let me rephrase it, no driver seems to be doing any kind of check on dev. Should I remove the asserts?